src/dmd/mars.d

```diff
@@ -817,9 +817,13 @@ version (NoMain) {} else
         {
             buf.writeEscapedMakePath(&params.exefile[0]);
         }
-        else if (params.lib && library)
+        else if (params.lib)
         {
-            buf.writeEscapedMakePath(library.getFilename());
+            const(char)[] libname = params.libname ? params.libname :
+                                                     FileName.forceExt(params.objfiles[0].toDString, target.lib_ext);
+            if (!FileName.absolute(libname))
```

Comment:
Does this logic need to be in sync with Library.setFilename to work correctly? I'm not exactly sure what's going on, but this looks like the bad kind of code duplication. Maybe let `emitMakeDeps` take a string instead of a library and pass `library.getFilename()` as an argument.

Reply:
I think that @thewilsonator wants to get rid of the dependency on `library`.

Reply:
From mars.d entirely?

Reply:
Yes, all of the use of `Library` can be moved into a function in glue, glue can be made mostly `private`, and glue will then become the sole importer of dmd.lib